homeassistant/components/notify/pushover.py

```diff
@@ -64,13 +89,87 @@ def send_message(self, message='', **kwargs):
             if target is not None:
                 data['device'] = target
 
             try:
-                self.pushover.send_message(message, **data)
+                self.pushover.send_message(
+                    message=message, attachment=file, **data)
+
             except ValueError as val_err:
                 _LOGGER.error(str(val_err))
             except RequestError:
                 _LOGGER.exception("Could not send pushover notification")
+            finally:
+                # Remove the tempfile if one was created
+                if file_data and file_data.get(ATTR_FILE_URL) and filename:
+                    os.remove(filename)
+
+    def load_file(self, url=None, local_path=None, username=None,
+                  password=None, auth=None):
+        """Load image/document/etc from a local path or URL."""
+        try:
+            # Load the file from URL
+            if url is not None:
+                # Check whether authentication parameters are provided
+                if username is not None and password is not None:
+                    # Use digest or basic authentication
+                    if ATTR_FILE_AUTH_DIGEST == auth:
+                        auth_ = HTTPDigestAuth(username, password)
+                    else:
+                        auth_ = HTTPBasicAuth(username, password)
+                    # Load file from URL with authentication
+                    response = requests.get(
+                        url, auth=auth_, timeout=CONF_TIMEOUT)
+                else:
+                    # Load file from URL without authentication
+                    response = requests.get(url, timeout=CONF_TIMEOUT)
+
+                # Make the request and raise an error if necessary
+                response.raise_for_status()
+
+                downloaded_file = (
+                    tempfile.NamedTemporaryFile(delete=False)
+                )
+
+                filename = downloaded_file.name
+                downloaded_file.write(response.content)
+
+                return filename
+
+            # Load the file from the filesystem
+            elif local_path is not None:
+                # Change the path if the file is in the local www directory
+                regex = re.compile('^/local/')
+                local_path = regex.sub(self._local_www_path + "/", local_path)
+
+                # Check whether path is whitelisted in configuration.yaml
+                if self._is_allowed_path(local_path):
+                    return local_path
+                _LOGGER.warning("'%s' is not secure to load data from!",
+                                local_path)
+            else:
+                _LOGGER.warning("Neither URL nor local path found in params!")
+
+        except requests.exceptions.RequestException as request_error:
+            _LOGGER.error("Can't load from url: %s", request_error)
+
+        except OSError as error:
+            _LOGGER.error("Can't load from url or local path: %s", error)
```

Comment:
Not allow. Do only exception handling around the function they can raise it.

Reply:
I'm sorry, I don't understand the request here.  Should we not be catching the OSError even though I believe that one could be raised?  Also, this was to be consistent with slack.py's load_file function.  I'm happy to change it, I just don't understand the request.